wdl-doc/src/lib.rs

```diff
@@ -0,0 +1,353 @@
+use wdl_ast::v1::MetadataValue;
+
+/// A WDL document.
+#[derive(Debug)]
+pub struct Document {
+    /// The name of the document.
+    ///
+    /// This is the filename of the document without the extension.
+    name: String,
+    /// The version of the document.
+    version: Version,
+    /// The Markdown preamble comments.
+    preamble: String,
+}
+
+impl Document {
+    /// Create a new document.
+    pub fn new(name: String, version: Version, preamble: String) -> Self {
+        Self {
+            name,
+            version,
+            preamble,
+        }
+    }
+
+    /// Get the name of the document.
+    pub fn name(&self) -> &str {
+        &self.name
+    }
+
+    /// Get the version of the document.
+    pub fn version(&self) -> String {
+        self.version.as_str().to_owned()
+    }
+
+    /// Get the preamble comments of the document.
+    pub fn preamble(&self) -> &str {
+        &self.preamble
+    }
+}
+
+impl Display for Document {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let document_name = content::Heading1::builder()
+            .text(self.name().to_owned())
+            .build();
+        let version = text_content::Paragraph::builder()
+            .text(format!("WDL Version: {}", self.version()))
+            .build();
+
+        let mut options = Options::empty();
+        options.insert(Options::ENABLE_TABLES);
+        options.insert(Options::ENABLE_STRIKETHROUGH);
+        let parser = Parser::new_ext(&self.preamble, options);
+        let mut preamble = String::new();
+        pulldown_cmark::html::push_html(&mut preamble, parser);
+
+        write!(f, "{}", document_name)?;
+        write!(f, "{}", version)?;
+        write!(f, "{}", preamble)
+    }
+}
+
+/// Fetch the preamble comments from a document.
+pub fn fetch_preamble_comments(document: AstDocument) -> String {
+    let comments = match document.version_statement() {
+        Some(version) => {
+            let comments = version
+                .keyword()
+                .syntax()
+                .preceding_trivia()
+                .map(|t| match t.kind() {
+                    wdl_ast::SyntaxKind::Comment => {
+                        match t.to_string().strip_prefix('#').unwrap().strip_prefix("# ") {
+                            Some(comment) => comment.to_string(),
+                            None => "".to_string(),
+                        }
+                    }
+                    wdl_ast::SyntaxKind::Whitespace => "".to_string(),
+                    _ => {
+                        panic!("Unexpected token kind: {:?}", t.kind())
+                    }
+                })
+                .collect::<Vec<_>>();
+            comments
+        }
+        None => {
+            vec![]
+        }
+    }
+    .join("\n");
+    comments
+}
+
+/// Generate HTML documentation for a workspace.
+pub async fn document_workspace(path: PathBuf) -> Result<()> {
+    if !path.exists() || !path.is_dir() {
```

Comment:
You're right. The first check is unnecessary:
> Returns true if the path exists on disk and is pointing at a directory.
>
> This function will traverse symbolic links to query information about the destination file.
>
> If you cannot access the metadata of the file, e.g. because of a permission error or broken symbolic links, this will return false.